features/admin.extensions.v1/components/administrators/pages/administrators.tsx

```diff
@@ -745,46 +752,49 @@ const CollaboratorsPage: FunctionComponent<CollaboratorsPageInterface> = (
      * Fetches the admin role id from database.
      */
     const getAdminRoleId = () => {
-        const searchData:SearchRoleInterface = {
-            filter: "displayName eq " + administratorConfig.adminRoleName,
-            schemas: [ "urn:ietf:params:scim:api:messages:2.0:SearchRequest" ],
-            startIndex: 0
-        };
-
-        searchRoleList(searchData)
-            .then((response: AxiosResponse) => {
-                if (response?.data?.Resources.length > 0) {
-                    let adminId: string = response?.data?.Resources[0]?.id;
-
-                    if (!legacyAuthzRuntime && response?.data?.Resources?.length > 1) {
-                        const filteredRoleList: RolesV2Interface[] = response?.data?.Resources?.filter(
-                            (role: RolesV2Interface) => role?.audience?.type === RoleAudienceTypes.APPLICATION);
-
-                        if (filteredRoleList?.length > 0) {
-                            adminId = filteredRoleList[0]?.id;
+        if (consoleId) {
+            const searchData:SearchRoleInterface = {
+                filter: "displayName eq " + administratorConfig.adminRoleName +
+                    " and audience.value eq " + consoleId,
+                schemas: [ "urn:ietf:params:scim:api:messages:2.0:SearchRequest" ],
+                startIndex: 0
+            };
+
+            searchRoleList(searchData)
+                .then((response: AxiosResponse) => {
+                    if (response?.data?.Resources.length > 0) {
+                        let adminId: string = response?.data?.Resources[0]?.id;
+
+                        if (!legacyAuthzRuntime && response?.data?.Resources?.length > 1) {
+                            const filteredRoleList: RolesV2Interface[] = response?.data?.Resources?.filter(
+                                (role: RolesV2Interface) => role?.audience?.type === RoleAudienceTypes.APPLICATION);
+
+                            if (filteredRoleList?.length > 0) {
+                                adminId = filteredRoleList[0]?.id;
+                            }
```

Comment:
adminId = filteredRoleList[0] ? filteredRoleList[0]?.id : adminId

Reply:
I have used the shorthand of the suggestion. This suggestion doesn't cover the required logic.